New Left Review.js

```diff
@@ -0,0 +1,257 @@
+{
+	"translatorID": "b97af278-52c1-4fc4-9c8b-5175c83941f2",
+	"label": "New Left Review",
+	"creator": "Bo An",
+	"target": "^https?://(www.)?newleftreview.org/issues/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2021-06-12 07:59:12"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+	Copyright © 2021 Bo An
+	This file is part of Zotero.
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	const article = "articles";
+	const isArticle = url.includes(article);
+	if (isArticle) {
+		return 'journalArticle';
+	}
+	else {
+		return 'multiple';
+	}
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), (items) => {
+			if (!items) {
+				return true;
+			}
+			const articles = [];
+			for (const i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+			return true;
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+function getSearchResults(doc) {
+	let items = {};
+	let found = false;
+
+	const articleLinks = ZU.xpath(doc, '//a[@class="article-link"]');
+	const titles = ZU.xpath(doc, '//span[@class="programme-note__title"]');
+	const authors = ZU.xpath(doc, '//span[@class="programme-note__author"]');
+
+	articleLinks.forEach((articleLink, articleIndex) => {
+		const href = articleLink.href;
+		let title = titles[articleIndex].innerText;
```

Comment:
And then replace this and the following line with this:

```js
let title = text(articleLink, '.programme-note__title');
let author = text(articleLink, '.programme-note__author');
```

It's a bit less fragile - even if, in the future, some page has an extra `.programme-note__title` that *isn't* inside a `.article-link`, or a `.article-link` has two authors, or some other weird mishap like that, restricting the search to children of the article link you're looking at will make sure that the arrays don't get out of alignment.

Reply:
thx so much! btw, what's this text() function?